Improve tag input keyboard navigation

SuaYoo opened the pull request:
Resolves https://github.com/webrecorder/browsertrix-cloud/issues/582:
- Re-focus on input after selection
- Delete tags with keyboard navigation
- Remove zero-width characters
- Fix being able to add tag with commas when pasting in Chrome

### Manual testing
Verify reports in [issue description](https://github.com/webrecorder/browsertrix-cloud/issues/582) are resolved as expected.

Manually tested in Chrome and Firefox, macOS

SuaYoo committed:
fix commas appended on chrome paste

SuaYoo committed:
bring input back to focus

SuaYoo committed:
remove zero-width characters

SuaYoo committed:
prevent tag from adding on concat

SuaYoo committed:
re-focus after removing tag

SuaYoo committed:
fix roving focus

SuaYoo committed:
close dropdown on tab out

SuaYoo committed:
wip

SuaYoo committed:
remove tag with delete

SuaYoo committed:
hide dropdown on end

SuaYoo committed:
fix input type

Shrinks99 commented:
One small fix, one possible fix?  Other than this, deleting tags with keyboard navigation is working nicely! :D

If all the tags are deleted out of the field with the delete key the field gets deselected.  This does not happen if the tags are deleted with the X button.  Ideally the field should stay selected?

https://user-images.githubusercontent.com/5672810/221742786-ee2f4ec6-6dfc-458c-9719-e90716c1e674.mov

---

Thoughts on selecting the next tag to the left if the user hits delete with no text in the box?  Should be a decent tradeoff between preventing accidental deletion & intuitively allowing people to hit the delete key to remove the next thing?

<img width="525" alt="Screenshot 2023-02-27 at 9 59 09 PM" src="https://user-images.githubusercontent.com/5672810/221742816-f9efe881-e351-4ec3-b634-67b22fe153ce.png">

stavares843 reviewed (commented):
neat 💥 

i've noticed that when we try to add a duplicated tag, there isn't a feedback for the user, in case we wanna improve that later

https://user-images.githubusercontent.com/29093946/221949945-ecf8de76-03b7-448d-9a00-3d188f0c632f.mov

Chrome, macOS

stavares843 reviewed (commented):
regarding the empty tag issue, is fixed but it has somewhat a glitch if you try to copy paste this [character](https://emojipedia.org/zero-width-joiner/), very minor though


https://user-images.githubusercontent.com/29093946/221950745-8f676a6d-34c4-4a5b-848d-0f0c1cfabbbc.mov


Chrome, macOS

stavares843 reviewed (commented):
<img width="503" alt="Captura de ecrã 2023-02-28, às 19 00 24" src="https://user-images.githubusercontent.com/29093946/221953098-ce360159-9a22-404c-9f79-072b78eaffb2.png">

Unrelated but the tags are clickable, is that for a future feature, example a filter?

Shrinks99 commented:
+1 to duplicate tag feedback, adding a single flash animation on the matching tag should be doable?  Unsure if it's for this issue or another, @SuaYoo Implement as you see fit. :)

SuaYoo committed:
select tag on backspace

SuaYoo committed:
fix pasting non-zero width

SuaYoo committed:
fix focusable tag

SuaYoo committed:
add shake animation

SuaYoo committed:
remove animation class on hide

SuaYoo commented:
> i've noticed that when we try to add a duplicated tag, there isn't a feedback for the user

Added a shake animation (sorry Hank, just saw your comment about flash, thoughts?):


https://user-images.githubusercontent.com/4672952/221991175-33646149-1118-46cb-b54e-82fd3a284bd0.mov

> If all the tags are deleted out of the field with the delete key the field gets deselected.
> glitch if you try to copy paste this [character](https://emojipedia.org/zero-width-joiner/)
> Unrelated but the tags are clickable

Fixed!

Shrinks99 reviewed (approved):
Shake gets the point across so works for me!

RE: Selecting on first item before delete, I like it! Seems intuitive to me! :D

Good to go! 🚀
